src/views/AddItem.jsx

```diff
@@ -1,22 +1,36 @@
 import { useState } from 'react';
 import { addItem } from '../api/firebase';
 
-export function AddItem({ listToken }) {
+export function AddItem({ listToken, data, setData }) {
 	const [daysUntilNextPurchase, setTimeFrame] = useState('7');
 	const [itemName, setItem] = useState('');
-	const [itemData, setData] = useState({});
 	const [error, setError] = useState(false);
+	const [duplicateError, setDuplicateError] = useState(false);
 	const [success, setSuccess] = useState(false);
 
+	// check for duplicate item in list
+	const isDuplicate = (data) =>
+		data.name.toLowerCase().replace(/ /g, '') ===
+		itemName.toLowerCase().replace(/ /g, '');
```

Comment:
regular expressions, nice, they are on my list of what I need to learn

Reply:
I think @Trevor-Rezac was using a site to generate regex and translate to english -- y'all should go over your process either here or @ demos

Reply:
niiiiice

Reply:
To be honest, @lindseyindev, we didnt even need to use the site for this. The space between the / / is looking for spaces in the word. the /g indicates the end of the regex expression. Michelle actually found this for us! Basically we just lowercase the string, remove any spaces, and compare!